features/cellular/framework/mux/cellular_mux.h

```diff
@@ -0,0 +1,561 @@
+/*
+ * Copyright (c) 2006-2017, Arm Limited and affiliates.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MUX_H
+#define MUX_H
+
+#include <stdbool.h>
+#include "FileHandle.h"
+#include "PlatformMutex.h"
+#include "Semaphore.h"
+#include "EventQueue.h"
+
+#define MUX_DLCI_INVALID_ID       0    /* Invalid DLCI ID. Used to invalidate MuxDataService object. */
+#define MUX_CRC_TABLE_LEN         256u /* CRC table length in number of bytes. */
+
+#ifndef MBED_CONF_MUX_DLCI_COUNT
```

Comment:
If you're adding mbed conf options, you need to actually have a mbed_lib.json, otherwise no-one can change them.

You can also put the defaults and help in there - no need to have a default here unless it needs to compile outside mbed OS.

Reply:
Managed by IOTCELL-570